### ISSUE TITLE
RUN "app.bas" doesn't work in user APK

### ISSUE THREAD

**Issue by mougino** [RFO-BASIC/Basic#184]
Found by user miki at http://rfobasic.freeforums.org/post19742.html#p19742
`RUN "app.bas"` does not work in a user APK.

So far the code that runs a bas file from asset in `Run.executeRUN()` is:
```Java
try { // or asset
  String assetPath = Basic.getAppFilePath(Basic.SOURCE_DIR, fileName);
  getAssets().openFd(assetPath); // exception if asset does not exist
  exists = true;
} catch (IOException ex) { }
```

There is a bug here: even if the asset exists, an exception will be thrown:
> **java.io.FileNotFoundException**: This file can not be opened as a file descriptor; it is probably compressed

A research pointed me to this interesting stackoverflow topic: http://stackoverflow.com/questions/6186866/java-io-filenotfoundexception-this-file-can-not-be-opened-as-a-file-descriptor

In a nutshell: depending on their extension some assets are compressed when creating the APK (typically *.bas) and some others are not (e.g. *.mp3) ; and trying to open a `FileDesriptor` on a compressed asset throws the exception mentioned above.

There is some evidence that you already knew (or at least heard about) this issue in `Run.getAssetType()`: you added a comment in a first try/catch using `AssetManager.openFd()` saying it `only works for some file extensions` ; then you wrote another try/catch using a more simple `InputStream = AssetManager.open()` (working on both uncompressed and compressed assets) this time with the comment `last ditch, should always work`.

**Solution:** to avoid doing the job twice, I suggest to replace the call to `AssetManager.openFd()` in `Run.executeRUN()` to a call to `getAssetType()`. How do you think??

I found 2 other calls to `AssetManager.openFd()`, concerning media files (that are supposed to be left untouched (uncompressed) when building the APK), so I think we're safe not to touch these methods: `private MediaPlayer getMP()` and `execute_SP_load()`.

**Comment by mougino** [comment]
I confirm that changing the existing code in `executeRUN()` from
```Java
else try { // or asset
  String assetPath = Basic.getAppFilePath(Basic.SOURCE_DIR, fileName);
  getAssets().openFd(assetPath); // exception if asset does not exist
  exists = true;
} catch (IOException ex) { }
```
to
```Java
else { // or asset
  String assetPath = Basic.getAppFilePath(Basic.SOURCE_DIR, fileName);
  if (getAssetType(assetPath) == "f") { exists = true; }  // it is a valid asset file
}
```
solves the problem.

I successfully compiled and built a BASIC! apk against which I built miki's app (2 bas running each other) in Quick APK and it works well.
I'll commit the fix to GitHub tonight.

**Comment by jMarcS** [comment]
Released in v01.86, 2015/04/11.
